includes/Order/Hooks.php

```diff
@@ -407,62 +410,50 @@ public function split_vendor_orders( $parent_order_id ) {
         $available_vendors  = [];
         $available_products = [];
 
-        if ( WC()->cart && ! WC()->cart->is_empty() ) {
-            foreach ( WC()->cart->get_cart() as $item ) {
-                $product_id           = $item['data']->get_id();
-                $available_vendors[]  = (int) dokan_get_vendor_by_product( $product_id, true );
-                $available_products[] = $product_id;
-            }
-        } else {
-            foreach ( $discount->get_items() as $item ) {
-                if ( ! isset( $item->product ) || ! $item->product instanceof \WC_Product ) {
-                    continue;
-                }
+	    foreach ( $discounts->get_items() as $item ) {
+		    if ( ! isset( $item->product ) || ! $item->product instanceof WC_Product ) {
+			    continue;
+		    }
 
-                $item_id = $item->product->get_id();
+		    $item_id = $item->product->get_id();
 
-                $available_vendors[]  = (int) dokan_get_vendor_by_product( $item_id, true );
-                $available_products[] = $item_id;
-            }
-        }
+		    $available_vendors[]  = (int) dokan_get_vendor_by_product( $item_id, true );
+		    $available_products[] = $item_id;
+	    }
 
         $available_vendors = array_unique( $available_vendors );
 
         if ( $coupon->is_type( 'fixed_cart' ) && count( $available_vendors ) > 1 ) {
             throw new Exception( esc_html__( 'This coupon is invalid for multiple vendors.', 'dokan-lite' ) );
         }
 
-        // Make sure applied coupon created by admin
-        if ( apply_filters( 'dokan_ensure_admin_have_create_coupon', $valid, $coupon, $available_vendors, $available_products ) ) {
-            return true;
-        }
-
-        if ( ! apply_filters( 'dokan_ensure_vendor_coupon', true ) ) {
-            return $valid;
-        }
-
-        // A coupon must be bound with a product
-        if ( ! dokan()->is_pro_exists() && count( $coupon->get_product_ids() ) === 0 ) {
-            throw new Exception( esc_html__( 'A coupon must be restricted with a vendor product.', 'dokan-lite' ) );
-        }
-
-        $coupon_id = $coupon->get_id();
-        $vendor_id = intval( get_post_field( 'post_author', $coupon_id ) );
-
-        if ( ! in_array( $vendor_id, $available_vendors, true ) ) {
-            return false;
-        }
+	    /**
+	     * Filter the validity of a coupon.
+	     *
+	     * @since DOKAN_SINCE
+	     *
+	     * @param boolean      $valid              The validity of the coupon.
+	     * @param WC_Coupon    $coupon             The coupon object.
+	     * @param array        $available_vendors  List of available vendors.
+	     * @param array        $available_products List of available products.
+	     * @param WC_Discounts $discounts          The discount object, which contains the order details.
+	     */
+	    if ( apply_filters( 'dokan_coupon_is_valid', $valid, $coupon, $available_vendors, $available_products, $discounts ) ) {
```

Comment:
Pls assign to `$valid` or return by applying the filter.

Reply:
vai, done